File: LGKotlin1/app/src/main/java/com/yotamarker/lgkotlin1/DSpeller.java
package com.yotamarker.lgkotlin1;


import java.util.ArrayList;

public class DSpeller extends AbsCmdReq implements Neuronable {
    private String param = "";
    private Boolean active;
    private  DISkillUtils diSkillUtil = new DISkillUtils();
    private TrgM1 TrgM1 = new TrgM1();
    @Override
    public void output(Neuron noiron) {
        switch (this.param) {
            case "about":
                param = "";
                noiron.algParts.add(
                        verbatimGorithm(new APVerbatim("I am", "the living grimoire", "I was created by", "moti barski")));
                return;
            case "creator":
                param = "";
                noiron.algParts.add(verbatimGorithm(new APVerbatim("I was created by", "moti barski")));
                return;
            case "hello":
                param = "";
                PlayGround playGround = new PlayGround();
                int temp = TrgM1.getMood();
                if (temp == 0) {
                    noiron.algParts.add(verbatimGorithm(new APVerbatim("good " + playGround.partOfDay())));
                } else {
                    noiron.algParts.add(verbatimGorithm(new APVerbatim("what ever")));
                }
                return;
            case "god mode":
                param = "";
                noiron.algParts.add(verbatimGorithm(new APVerbatim("ainzbuff")));
                return;
            case "do you like me":
                param = "";
                noiron.algParts.add(verbatimGorithm(new APVerbatim("pogchamp")));
                return;
            case "kiss":
                param = "";
                noiron.algParts.add(verbatimGorithm(new APVerbatim("kiss")));
                return;
            case "hadouken":
                param = "";
                noiron.algParts.add(diSkillUtil.verbatimGorithm("spell",new APVerbatim("hadouken")));
                return;
            case "shouryuken":
                param = "";
                noiron.algParts.add(diSkillUtil.verbatimGorithm("spell",new APVerbatim("shouryuken")));
                return;
            default:
                break;
        }
        if (this.active) {
            APSpell maho = new APSpell(this.param);
            ArrayList<AbsAlgPart> algParts1 = new ArrayList<>();
            algParts1.add(maho);
            Algorithm algorithm = new Algorithm("spell", this.param, algParts1);
            noiron.algParts.add(algorithm);
        }
    }

    @Override
    public void input(String ear, String skin, String eye) {
        TrgM1.trigger(ear, skin, eye);
        switch (ear) {
            case "what is the time":
            case "what is the date":
            case "what is the year":
            case "current seconds":
            case "current minutes":
            case "current hour":
            case "which day is it":
            case "greet":
                active = true;
                this.param = ear;
                break;
            case "what are you":
                this.param = "about";
                break;
            case "hello":
                this.param = "hello";
                break;
            case "who made you":
                this.param = "creator";
                break;
            case "god mode":
                this.param = "god mode";
                break;
            case "do you like me":
                this.param = "do you like me";
                break;
            case "give me a kiss":
                this.param = "kiss";
                break;
            case "hadouken": case"hadoken":
                this.param = "shouryuken";
                break;
            case "shouryuken": case"shoryuken":
                this.param = "hadouken";
                break;
            default:
                active = false;
                break;

        }
    }

    private Algorithm verbatimGorithm(AbsAlgPart itte) {
        // returns a simple algorithm for saying sent parameter
        // AbsAlgPart itte = new APVerbatim("I am");
        String representation = "about";
        ArrayList<AbsAlgPart> algParts1 = new ArrayList<>();
        algParts1.add(itte);
        Algorithm algorithm = new Algorithm("about", representation, algParts1);
        return algorithm;
    }
}

File: livingrimoire start here/LivingGrimoire Kotlin/skills AlgParts and support classes/DSpeller.kt
package com.yotamarker.lgkotlinfull.skills

import com.yotamarker.lgkotlinfull.LGCore.*
import java.util.*


class DSpeller : AbsCmdReq() {
    private var param = ""
    private var active: Boolean? = null
    private val diSkillUtil = DISkillUtils()
    private val TrgM1: TrgM1 = TrgM1()
    override fun output(noiron: Neuron) {
        when (param) {
            "about" -> {
                param = ""
                noiron.algParts.add(
                        verbatimGorithm(APVerbatim("I am", "the living grimoire", "I was created by", "moti barski")))
                return
            }
            "creator" -> {
                param = ""
                noiron.algParts.add(verbatimGorithm(APVerbatim("I was created by", "moti barski")))
                return
            }
            "hello" -> {
                param = ""
                val playGround = PlayGround()
                val temp: Int = TrgM1.getMood()
                if (temp == 0) {
                    noiron.algParts.add(verbatimGorithm(APVerbatim("good " + playGround.partOfDay())))
                } else {
                    noiron.algParts.add(verbatimGorithm(APVerbatim("what ever")))
                }
                return
            }
            "god mode" -> {
                param = ""
                noiron.algParts.add(verbatimGorithm(APVerbatim("ainzbuff")))
                return
            }
            "hadouken" -> {
                param = ""
                noiron.algParts.add(diSkillUtil.verbatimGorithm("spell", APVerbatim("hadouken")))
                return
            }
            "shouryuken" -> {
                param = ""
                noiron.algParts.add(diSkillUtil.verbatimGorithm("spell", APVerbatim("shouryuken")))
                return
            }
            else -> {
            }
        }
        if (active!!) {
            val maho = APSpell(param)
            val algParts1 = ArrayList<AbsAlgPart>()
            algParts1.add(maho)
            val algorithm = Algorithm("spell", param, algParts1)
            noiron.algParts.add(algorithm)
        }
    }

    override fun input(ear: String, skin: String, eye: String) {
        TrgM1.trigger(ear, skin, eye)
        when (ear) {
            "what is the time", "what is the date", "what is the year", "current seconds", "current minutes", "current hour", "which day is it", "greet" -> {
                active = true
                param = ear
            }
            "what are you" -> param = "about"
            "hello" -> param = "hello"
            "who made you" -> param = "creator"
            "god mode" -> param = "god mode"
            "hadouken", "hadoken" -> param = "shouryuken"
            "shouryuken", "shoryuken" -> param = "hadouken"
            else -> active = false
        }
    }

    private fun verbatimGorithm(itte: AbsAlgPart): Algorithm {
        // returns a simple algorithm for saying sent parameter
        // AbsAlgPart itte = new APVerbatim("I am");
        val representation = "about"
        val algParts1 = ArrayList<AbsAlgPart>()
        algParts1.add(itte)
        return Algorithm("about", representation, algParts1)
    }
}